docs/docs/08.1-more-about-refs.md

```diff
@@ -1,75 +1,76 @@
+```js
+var myComponentElement = <MyComponent />; // This is just a ReactElement.
+
+// Some code here...
+
+var myComponentInstance = React.render(myComponentElement, myContainer);
+myComponentInstance.doSomething();
+```
 
 > Note:
 >
-> Remember, what you return from `render()` is not your *actual* rendered children instances. What you return from `render()` is merely a *description* of the children instances in your component's sub-hierarchy at a particular moment in time.
+> This should only ever be used at the top level. Inside components, let your `props` and `state` handle communication with child components, or use one of the other methods of getting a ref (string attribute or callbacks).
+
 
+## The ref Callback Attribute
 
-This means that you should never "hold onto" something that you return from `render()` and then expect it to be anything meaningful.
+React supports a very special attribute that you can attach to any component.  The `ref` attribute can be a callback function, and this callback will be executed immediately after the component is mounted.  The referenced component will be passed in as a parameter, and the callback function may use the component immediately, or save the reference for future use (or both).
 
-```javascript
-  // antipattern: This won't work.
+It's as simple as adding a `ref` attribute to anything returned from `render`:
+
+```html
   render: function() {
-    var myInput = <input />;          // I'm going to try to call methods on this
-    this.rememberThisInput = myInput; // input at some point in the future! YAY!
     return (
-      <div>
-        <div>...</div>
-        {myInput}
-      </div>
+      <TextInput
+        ref={function(input) {
+          if (input != null) {
+            input.focus();
+          }
+        }} />
     );
-  }
+  },
 ```
+or using an ES6 arrow function:
 
-In this example, the `<input />` is merely a *description* of an `<input />`. This description is used to create a *real* **backing instance** for the `<input />`.
+```html
+  render: function() {
+    return <TextInput ref={(c) => this._input = c} />;
+  },
+  componentDidMount: function() {
+    this._input.focus();
+  },
+```
 
-So how do we talk to the *real* backing instance of the input?
+  Note that when the referenced component is unmounted and whenever the ref changes, the old ref will be called with `null` as an argument. This prevents memory leaks in the case that the instance is stored, as in the first example. Note that when writing refs with inline function expressions as in the examples here, React sees a different function object each time so on every update, ref will be called with `null` immediately before it's called with the component instance.
```

Comment:
> Note that… Note that…

Makes for an awkward paragraph. Maybe "Also note…"

Also, should these be the blockquote Note: format?

Also also, you have whitespace at the beginning of this paragraph (and the one below it).